core/src/mill/modules/Jvm.scala

```diff
@@ -61,17 +61,19 @@ object Jvm {
   }
 
 
-
+  /**
+    * run `body` with a isolated classloader
+    */
   def inprocess[T](classPath: Agg[Path],
-                   classLoaderOverrideSbtTesting: Boolean,
+                   classLoaderOverrideSbtTesting: Option[ClassLoader],
                    body: ClassLoader => T): T = {
-    val cl = if (classLoaderOverrideSbtTesting) {
-      val outerClassLoader = getClass.getClassLoader
+    val cl = if (classLoaderOverrideSbtTesting.isDefined) {
+      val outerClassLoader = classLoaderOverrideSbtTesting.get
```

Comment:
Maybe we could pattern match or fold `Option` here, rather than calling `.isDefined` and `.get` on it?